### DISCUSSION TITLE
async progressbar with System.Net.Webclient.DownloadFileTaskAsync

### DISCUSSION THREAD

**Discussion by bananaboy139** [spectreconsole/spectre.console#474] [Q&A]
How do I use AnsiConsole.Progress().StartAsync() with System.Net.Webclient.DownloadFileTaskAsync()?
I can't get the progress to work, it starys at 0.

            AnsiConsole.Progress()
                .AutoClear(false)
                .HideCompleted(false)
                .Columns(new ProgressColumn[]
                {
                    new ProgressBarColumn(),
                    new PercentageColumn(),
                    new RemainingTimeColumn(),
                    new SpinnerColumn()
                })
                .StartAsync(async ctx =>
                {
                    var task = ctx.AddTask("Download [link=" + URL + "]" + filename + "[/]");
                    await Dow.DownloadAsync(LINK, task);
                });

        private void Wc_DownloadProgressChanged(object sender, DownloadProgressChangedEventArgs e)
        {
            task.MaxValue = e.TotalBytesToReceive;
            task.Increment(e.BytesReceived);
        }

Dow.DownloadAsync is just a function to setup the webcllient and start the DownloadFileTaskAsync() method

edit:
repo: https://github.com/bananaboy139/MDownloader
warning, my code is terrible

**Comment by patriksvensson** [comment]
Could you post a link to a repository containing a minimal reproducable example, and I'll take a look at it.

**Reply by bananaboy139** [reply]
repo: https://github.com/bananaboy139/MDownloader
warning, my code is terrible
